File: web/lab2/src/main/java/com/radiad/lab2/Point.java
package com.radiad.lab2;

public class Point {
    private double x;
    private double y;
    private String stringX;
    private String stringY;
    private boolean isValid;
    private String warning;

    public Point(String stringX, String stringY) {
        this.stringX = stringX;
        this.stringY = stringY;
        this.validatePoint();
    }

    private boolean validateX() {
        try {
            x = Double.parseDouble(stringX);
            if (!(-5 <= x && x <= 3)) {
                warning = "неверный диапазон x";
                return false;
            }
        } catch (NumberFormatException e) {
            warning = "x должен быть числом";
            return false;
        }
        return true;
    }

    private boolean validateY() {
        try {
            y = Double.parseDouble(stringY);
            if (!(-5 <= y && y <= 3)) {
                warning = "неверный диапазон y";
                return false;
            }
        } catch (NumberFormatException e) {
            warning = "y должен быть числом";
            return false;
        }
        return true;
    }

    private void validatePoint() {
        this.isValid = this.validateX() && this.validateY();
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public boolean isValid() {
        return isValid;
    }

    public String getWarning() {
        return warning;
    }
}

File: web/lab1/server/Point.php
<?php

class Point {
    private $x;
    private $y;
    private bool $isValid;
    private array $warnings;

    public function __construct($x, $y) {
        $this->x = $x;
        $this->y = $y;
        $this->validatePoint();
    }

    private function validateX(): bool
    {
        if (!is_numeric($this->x)){
            $this->warnings[] = "x должен быть числом";
            return false;
        }
        if ((double) $this->x - (int) $this->x){
            $this->warnings[] = "x должен быть целым числом";
            return false;
        }
        if (!(-5 <= $this->x && $this->x <= 3)){
            $this->warnings[] = "неверный диапазон x";
            return false;
        }
        $this->x = (int) $this->x;
        return true;
    }

    private function validateY(): bool
    {
        if (!is_numeric($this->y)){
            $this->warnings[] = "y должен быть числом";
            return false;
        }
        if (!(-5 <= $this->y && $this->y <= 3)){
            $this->warnings[] = "неверный диапазон y";
            return false;
        }
        return true;
    }

    private function validatePoint(): void
    {
        $this->isValid = $this->validateX() && $this->validateY();
    }

    public function getX()
    {
        return $this->x;
    }

    public function getY()
    {
        return $this->y;
    }

    public function isValid(): bool
    {
        return $this->isValid;
    }
    public function getWarnings(): array
    {
        return $this->warnings;
    }
}